scripts/issue/71/consolidate_badger.py

```diff
@@ -0,0 +1,145 @@
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import registry
+from brownie import interface
+from rich.console import Console
+
+CONSOLE = Console()
+
+DEV_MULTISIG = registry.eth.badger_wallets.dev_multisig
+OLD_OPS_MULTISIG = registry.eth.badger_wallets.ops_multisig_old
+TREASURY_VAULT = registry.eth.badger_wallets.treasury_vault_multisig
+
+BADGER_GEYSER = registry.eth.badger_geyser
+
+def main(upgrade="true", simulation="false"):
+    badger_token = interface.IBadger(registry.eth.treasury_tokens.BADGER)
+
+    old_multi = GreatApeSafe(OLD_OPS_MULTISIG)
+
+    old_devProxyAdmin = interface.IProxyAdmin(
+        registry.eth.badger_wallets.opsProxyAdmin_old,
+        owner=old_multi.account
+    )
+    dao_treasury = interface.ISimpleTimelockWithVoting(
+        registry.eth.badger_wallets.DAO_treasury,
+        owner=old_multi.account
+    )
+
+    if upgrade == "true":
+        ## == Upgrade DAO_Treasury to new logic from the opsMultisig_old == ##
+
+        # Save storage variabls for later
+        treasury_balance = badger_token.balanceOf(dao_treasury.address)
+        token = dao_treasury.token()
+        releaseTime = dao_treasury.releaseTime()
+        beneficiary = dao_treasury.beneficiary()
+        
+        # Upgrades logic and verifies storage
+        new_logic = registry.eth.logic.SimpleTimelockWithVoting
+        old_devProxyAdmin.upgrade(dao_treasury.address, new_logic)
+
+        assert treasury_balance == badger_token.balanceOf(dao_treasury.address)
+        assert token == dao_treasury.token()
+        assert releaseTime == dao_treasury.releaseTime()
+        assert beneficiary == dao_treasury.beneficiary()
+        # Beneficiary is currently the DAO_Agent
+        assert dao_treasury.beneficiary() == registry.eth.badger_wallets.fees
+
+        # Nonces are messed up on this multi - next in line is 78
+        old_multi.post_safe_tx(replace_nonce=78)
```

Comment:
I am thinking that in this msig if the first nonce is going to be anyway rejected (*76*), perhaps is worthy to directly replace nonce 76, instead of paying for a rejection, thoughts?

Reply:
Good idea, I don't see the harm. I confirmed that 76 is obsolete and should be rejected. Will make the change.